pkgs/data/themes/catppuccin-gtk/default.nix

```diff
@@ -4,64 +4,81 @@
 , gtk3
 , gnome-themes-extra
 , gtk-engine-murrine
+, python3
 , sassc
-, tweaks ? [ ]
+, accent ? "blue"
 , size ? "standard"
+, tweaks ? [ ]
+, variant ? "frappe"
 }:
 let
+  validAccents = [ "blue" "flamingo" "green" "lavender" "maroon" "mauve" "peach" "pink" "red" "rosewater" "sapphire" "sky" "teal" "yellow" ];
   validSizes = [ "standard" "compact" ];
-  validTweaks = [ "nord" "dracula" "black" "rimless" "normal" ];
-
-  unknownTweaks = lib.subtractLists validTweaks tweaks;
-  illegalMix = (lib.elem "nord" tweaks) && (lib.elem "dracula" tweaks);
+  validTweaks = [ "black" "rimless" "normal" ];
+  validVariants = [ "latte" "frappe" "macchiato" "mocha" ];
 
-  assertIllegal = lib.assertMsg (!illegalMix) ''
-    Tweaks "nord" and "dracula" cannot be mixed. Tweaks: ${toString tweaks}
+  assertAccent = lib.assertMsg (lib.elem accent validAccents) ''
+    You entered an invalid accent: ${accent}
+    Valid accents are: ${toString validAccents}
   '';
 
   assertSize = lib.assertMsg (lib.elem size validSizes) ''
-    You entered a wrong size: ${size}
+    You entered an invalid size: ${size}
     Valid sizes are: ${toString validSizes}
   '';
 
-  assertUnknown = lib.assertMsg (unknownTweaks == [ ]) ''
-    You entered wrong tweaks: ${toString unknownTweaks}
+  unknownTweaks = lib.subtractLists validTweaks tweaks;
+  assertTweaks = lib.assertMsg (unknownTweaks == [ ]) ''
+    You entered invalid tweaks: ${toString unknownTweaks}
     Valid tweaks are: ${toString validTweaks}
   '';
+
+  assertVariants = lib.assertMsg (lib.elem variant validVariants) ''
+    You entered an invalid variant: ${variant}
+    Valid variants are: ${toString validVariants}
+  '';
 in
 
-assert assertIllegal;
+assert assertAccent;
 assert assertSize;
-assert assertUnknown;
+assert assertTweaks;
+assert assertVariants;
 
 stdenvNoCC.mkDerivation rec {
   pname = "catppuccin-gtk";
-  version = "0.2.7";
+  version = "0.4.0";
 
   src = fetchFromGitHub {
     repo = "gtk";
     owner = "catppuccin";
-    rev = "v-${version}";
-    sha256 = "sha256-oTAfURHMWqlKHk4CNz5cn6vO/7GmQJM2rXXGDz2e+0w=";
+    rev = "v${version}";
+    sha256 = "sha256-OFJeuwRBjqIGdP1vPX431DtpkPYZPgvUw3KFBrIsWnE=";
+    fetchSubmodules = true;
```

Comment:
Why not use our colloid?

Reply:
uhh what's that supposed to mean

Reply:
Why use the vendored copy of colloid (which is included in the repo as a submodule) over our copy is what he's asking.

Reply:
Yeah, The copy of colloid really doesn't matter much, just that it's copied over to the `colloid/` directory.

Reply:
I've tried doing this
```nix
  postUnpack = ''
    rmdir source/colloid || true
    cp -r ${colloid-gtk-theme.src} source/colloid
  '';
```
but it gives me this error
```nix
error: builder for '/nix/store/1sk85q8m7azmfs082m26x55cwddaci81-catppuccin-gtk-0.4.0.drv' failed with exit code 4;
       last 7 log lines:
       > unpacking sources
       > unpacking source archive /nix/store/pp6kilw980ld5b2b3ynd1p6a7q8mqh2a-source
       > source root is source
       > patching sources
       > patching script interpreter paths in colloid/clean-old-theme.sh colloid/install.sh
       > colloid/clean-old-theme.sh: interpreter directive changed from "#! /usr/bin/env bash" to "/nix/store/dsd5gz46hdbdk2rfdimqddhq6m8m8fqs-bash-5.1-p16/bin/bash"
       > sed: couldn't open temporary file colloid/sedUIbA0Q: Permission denied
```

Reply:
chmod +x the script

Reply:
Maybe upstream wants a specific revision of colloid to build the catppuccin gtk theme.

Reply:
> chmod +x the install script

```
› l /tmp/nix-build-catppuccin-gtk-0.4.0.drv-4/source/colloid/install.sh
.r-xr-xr-x 19k nixbld1 17 Dec 14:12  /tmp/nix-build-catppuccin-gtk-0.4.0.drv-4/source/colloid/install.sh
```

It's already executable, so it must be something else.

Reply:
> Maybe upstream wants a specific revision of colloid to build the catppuccin gtk theme.

I've thought about it as well, maybe @rubyowo knows more.

Reply:
It doesn't need any specific revision to work

Reply:
It seems like it works if I `chmod -R +w` the entire colloid source. Is this good enough?

Reply:
I think it's the only way to solve the error you got after reading the py code, but I guess @SuperSandro2000 would know if there's any other better solution